File: Seafood Guide/Top 10/Table View Cell/NewsViewController.m
//
//  NewsViewController.m
//  Seafood Guide
//
//  Created by Jon Brown on 9/2/14.
//  Copyright (c) 2014 Jon Brown Designs. All rights reserved.
//

#import "NewsViewController.h"
#import "TableViewCellNews.h"
#import "RXMLElement.h"
#import "DetailNewsViewController.h"
#import "config.h"
#import <QuartzCore/QuartzCore.h>

@interface NewsViewController ()

@end

@implementation NewsViewController
@synthesize NewsTbView,numArray,imageArray,descArray,titleArray;

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        
    }
    return self;
}


- (void)viewDidLoad
{
    
    [super viewDidLoad];

    self.numArray = [[NSMutableArray alloc] init];
    self.imageArray = [[NSMutableArray alloc]init];
    self.descArray = [[NSMutableArray alloc]init];
    self.titleArray = [[NSMutableArray alloc]init];
    
    NewsTbView.delegate = self;
    NewsTbView.dataSource = self;
    
    [self refresh];

}

-(void)refresh
{
    
    RXMLElement *rootXML = [RXMLElement elementFromXMLFile:@"ios-news.xml"];
    
    RXMLElement *rxmlNews = [rootXML child:@"news"];
    
    NSArray *rxmlIndividualNew = [rxmlNews children:@"new"];
        
    for (int i=0; i<rxmlIndividualNew.count; i++) {
        NSString *imgUrl = [NSString stringWithFormat:@"%@" , [[rxmlIndividualNew objectAtIndex:i] child:@"imageurl"].text];
        NSString *num = [NSString stringWithFormat:@"%@" , [[rxmlIndividualNew objectAtIndex:i] child:@"linknews"].text];
        UIImage *img1 = [UIImage imageNamed:imgUrl];
        
        NSString *title = [NSString stringWithString:[[rxmlIndividualNew objectAtIndex:i] child:@"titlenews"].text];
        NSString *desc = [NSString stringWithString:[[rxmlIndividualNew objectAtIndex:i] child:@"descnews"].text];
        
        [numArray addObject:num];
        [imageArray addObject:img1];
        [titleArray addObject:title];
        [descArray addObject:desc];
    }
    
    [NewsTbView reloadData];
    
}

#pragma mark NSURLConnection Delegate Methods

- (void)connection:(NSURLConnection *)connection didReceiveResponse:(NSURLResponse *)response {

    _responseData = [[NSMutableData alloc] init];
}

- (void)connection:(NSURLConnection *)connection didReceiveData:(NSData *)data {
    [_responseData appendData:data];
}

- (NSCachedURLResponse *)connection:(NSURLConnection *)connection
                  willCacheResponse:(NSCachedURLResponse*)cachedResponse {
    return nil;
}

- (void)connectionDidFinishLoading:(NSURLConnection *)connection {

    RXMLElement *rootXML = [RXMLElement elementFromXMLFile:@"ios-news.xml"];
    
    RXMLElement *rxmlNews = [rootXML child:@"news"];
    
    NSArray *rxmlIndividualNew = [rxmlNews children:@"new"];
    
    
    for (int i=0; i<rxmlIndividualNew.count; i++) {
        
        NSString *imgUrl = [NSString stringWithFormat:@"%@" , [[rxmlIndividualNew objectAtIndex:i] child:@"imageurl"].text];
        
        UIImage *img1 = [UIImage imageNamed:imgUrl];
        
        NSString *title = [NSString stringWithString:[[rxmlIndividualNew objectAtIndex:i] child:@"titlenews"].text];
        NSString *desc = [NSString stringWithString:[[rxmlIndividualNew objectAtIndex:i] child:@"descnews"].text];
        
        [imageArray addObject:img1];
        [titleArray addObject:title];
        [descArray addObject:desc];
    }
    
    
    
    [NewsTbView reloadData];
    
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
}

- (void)connection:(NSURLConnection *)connection didFailWithError:(NSError *)error {

}


- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
}



#pragma mark - Table View

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return imageArray.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *CellIdentifier = @"CellIndent";
    TableViewCellNews *cell = (TableViewCellNews *) [tableView dequeueReusableCellWithIdentifier:CellIdentifier];
    NSString *CellName;
    
    CellName = @"TableViewCellNews";
    
    if (cell == nil) {
        
        UIViewController *c = [[UIViewController alloc] initWithNibName:CellName bundle:nil];
        
        cell = (TableViewCellNews *)c.view;
    }
    
    cell.textLabel.numberOfLines = 0;
    cell.numb.text = [numArray objectAtIndex:indexPath.row];
    cell.image.image = [imageArray objectAtIndex:indexPath.row];
    cell.image.layer.cornerRadius = 5;
    cell.image.clipsToBounds = YES;
    cell.image.layer.borderColor=[[UIColor grayColor] CGColor];
    cell.image.layer.borderWidth=1.2;
    
    cell.circle.layer.cornerRadius=15;
    cell.circle.clipsToBounds=YES;
    cell.circle.layer.borderColor=[[UIColor grayColor] CGColor];
    cell.circle.layer.borderWidth=2.0;
    
    cell.text.text = [titleArray objectAtIndex:indexPath.row];
    cell.image.contentMode = UIViewContentModeScaleToFill;
    [cell setSelectionStyle:UITableViewCellSelectionStyleNone];
    [cell.image setAlpha:0];
    [cell.imagetitle setAlpha:0];
    [cell.text setAlpha:0];
    [UIView beginAnimations:@"ResizeAnimation" context:NULL];
    [UIView setAnimationDuration:0.5f];
    [cell.image setAlpha:1];
    [cell.text setAlpha:1];
    [cell.imagetitle setAlpha:1];
    [UIView commitAnimations];
    
    return cell;
}


- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    

    
DetailNewsViewController *controller = [self.storyboard instantiateViewControllerWithIdentifier:@"detailview"];
    
    if (_detailNewsViewController == nil) {
        
        NSString * storyboardName = @"Main_iPhone";
        UIStoryboard *storyboard = [UIStoryboard storyboardWithName:storyboardName bundle: nil];
        self.detailNewsViewController = [storyboard instantiateViewControllerWithIdentifier:@"detailview"];

    }

    
    controller.image = [imageArray objectAtIndex:indexPath.row];
    controller.lblTitle = [titleArray objectAtIndex:indexPath.row];
    controller.txtProject = [descArray objectAtIndex:indexPath.row];
    self.navigationItem.backBarButtonItem = [[UIBarButtonItem alloc]initWithTitle:@"Back" style:UIBarButtonItemStylePlain target:nil action:nil];
    [self.navigationController pushViewController:controller animated:YES];
    

}

-(UIButton*)createButtonWithFrame:(CGRect)frame andLabel:(NSString*)label
{
    
    UIButton *button = [[UIButton alloc] initWithFrame:frame];
    
    [button setTitleShadowColor:[UIColor blackColor] forState:UIControlStateNormal];
    button.titleLabel.shadowOffset = CGSizeMake(0, -1);
    button.titleLabel.font = [UIFont boldSystemFontOfSize:18];
    
    [button setTitle:label forState:UIControlStateNormal];
    [button setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    
    [[button layer] setBorderWidth:1.0f];
    [[button layer] setBorderColor:[UIColor whiteColor].CGColor];
    [[button layer] setMasksToBounds:YES];
    [[button layer] setCornerRadius:4.0]; 
    [[button layer] setBorderWidth:1.0];
    
    return button;
}


@end

File: Seafood Guide/Top 10/Table View Cell/NewsViewController.swift
//  Converted to Swift 5.1 by Swiftify v5.1.26565 - https://objectivec2swift.com/
//
//  NewsViewController.swift
//  Seafood Guide
//
//  Created by Jon Brown on 9/2/14.
//  Copyright (c) 2014 Jon Brown Designs. All rights reserved.
//

import QuartzCore
import UIKit
import SWXMLHash

class NewsViewController: UIViewController, UITableViewDelegate, UITableViewDataSource, UINavigationBarDelegate, UINavigationControllerDelegate {
    var objects: [AnyHashable] = []
    var responseData: Data?
    var backButton: UIButton?

    @IBOutlet var newsTbView: UITableView!
    var numArray: [String] = []
    var imageArray: [UIImage] = []
    var titleArray: [String] = []
    var descArray: [String] = []
    var detailNewsViewController: DetailNewsViewController?

    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        
        navigationController?.navigationBar.barStyle = .default
        let appearance = UINavigationBarAppearance()
        appearance.titleTextAttributes = [.foregroundColor: UIColor(named: "Text Color")!]
        appearance.largeTitleTextAttributes = [.foregroundColor: UIColor(named: "Text Color")!]
        appearance.backgroundColor = .clear

        navigationItem.standardAppearance = appearance
        navigationItem.scrollEdgeAppearance = appearance
        
        
        numArray = []
        imageArray = []
        descArray = []
        titleArray = []
        newsTbView.delegate = self
        newsTbView.dataSource = self
        refresh()
    }

    func refresh() {

        
        guard let xml = Bundle.main.path(forResource: "ios-news", ofType: "xml") else {
            print("File not found")
            return
        }
        let data = try! Data(contentsOf: URL(fileURLWithPath: xml))
        let rootXMLAbout1 = XMLHash.parse(data)
        let rxmlIndividualNew = rootXMLAbout1["xml"]["news"]["new"]

        
        for elem in rxmlIndividualNew.all {
            let imgURL = elem["imageurl"].element!.text
            let img1 = UIImage(named: imgURL)
            
            
            let num = elem["linknews"].element!.text
            let desc = elem["descnews"].element!.text
            let title = elem["titlenews"].element!.text
            
            numArray.append(num)
                if let img1 = img1 {
                    imageArray.append(img1)
                }
            
            titleArray.append(title)
            descArray.append(desc)
            
        }
        

        newsTbView.reloadData()

    }

// MARK: NSURLConnection Delegate Methods
    func connection(_ connection: NSURLConnection, didReceive response: URLResponse) {

        responseData = Data()
    }

    func connection(_ connection: NSURLConnection, didReceive data: Data) {
        responseData?.append(data)
    }

    func connection(_ connection: NSURLConnection, willCacheResponse cachedResponse: CachedURLResponse) -> CachedURLResponse? {
        return nil
    }

    func connectionDidFinishLoading(_ connection: NSURLConnection) {

        guard let xml = Bundle.main.path(forResource: "ios-news", ofType: "xml") else {
            print("File not found")
            return
        }
        let data = try! Data(contentsOf: URL(fileURLWithPath: xml))
        let rootXMLAbout1 = XMLHash.parse(data)
        let rxmlIndividualNew = rootXMLAbout1["xml"]["news"]["new"]


        for elem in rxmlIndividualNew.all {
            let imgURL = elem["imageurl"].element!.text
            let img1 = UIImage(named: imgURL)
            let desc = elem["descnews"].element!.text
            let title = elem["titlenews"].element!.text
            
            if let img1 = img1 {
                            imageArray.append(img1)
                        }
            
            titleArray.append(title)
            descArray.append(desc)
            
        }



        newsTbView.reloadData()

    }

    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
    }

    func connection(_ connection: NSURLConnection, didFailWithError error: Error) {
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

// MARK: - Table View
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }

    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return imageArray.count
    }

    static let tableViewCellIdentifier = "CellIndent"

    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        var cell = tableView.dequeueReusableCell(withIdentifier: NewsViewController.tableViewCellIdentifier) as? TableViewCellNews
        var CellName = ""

        CellName = "TableViewCellNews"

        if cell == nil {

            let c = UIViewController(nibName: CellName, bundle: nil)

            cell = c.view as? TableViewCellNews
        }

        cell?.numberLabel.text = numArray[indexPath.row]
        cell?.newsImageView.image = imageArray[indexPath.row]
        cell?.newsImageView.image = imageArray[indexPath.row]
        cell?.newsImageView.layer.cornerRadius = 5
        cell?.newsImageView.clipsToBounds = true
        cell?.newsImageView.layer.borderColor = UIColor.gray.cgColor
        cell?.newsImageView.layer.borderWidth = 1.2

        cell?.circle.layer.cornerRadius = 15
        cell?.circle.clipsToBounds = true
        cell?.circle.layer.borderColor = UIColor.gray.cgColor
        cell?.circle.layer.borderWidth = 2.0

        cell?.textNewsLabel.text = titleArray[indexPath.row]
        cell?.newsImageView.contentMode = .scaleToFill
        cell?.selectionStyle = UITableViewCell.SelectionStyle.none
        cell?.newsImageView.alpha = 0
        cell?.textNewsLabel.alpha = 0
        cell?.textNewsLabel.alpha = 0
        cell?.newsImageView.alpha = 1
        cell?.textNewsLabel.alpha = 1
        cell?.newsImageView.alpha = 1

        return cell!
    }

    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let controller = self.storyboard?.instantiateViewController(withIdentifier: "detailview") as? DetailNewsViewController
        controller?.image = imageArray[indexPath.row]
        controller?.lblTitle = titleArray[indexPath.row]
        controller?.txtProject = descArray[indexPath.row]
        navigationItem.backBarButtonItem = UIBarButtonItem(title: "Back", style: .plain, target: nil, action: nil)
        if let controller = controller {
            navigationController?.pushViewController(controller, animated: true)
        }
    }

    func createButton(withFrame frame: CGRect, andLabel label: String?) -> UIButton? {

        let button = UIButton(frame: frame)

        button.setTitleShadowColor(UIColor.black, for: .normal)
        button.titleLabel?.shadowOffset = CGSize(width: 0, height: -1)
        button.titleLabel?.font = UIFont.boldSystemFont(ofSize: 18)

        button.setTitle(label, for: .normal)
        button.setTitleColor(UIColor.white, for: .normal)

        button.layer.borderWidth = 1.0
        button.layer.borderColor = UIColor.white.cgColor
        button.layer.masksToBounds = true
        button.layer.cornerRadius = 4.0
        button.layer.borderWidth = 1.0

        return button
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
}
